state/graphql-sender/src/main/java/com/expediagroup/streamplatform/streamregistry/state/graphql/DefaultApolloClientFactory.java

```diff
@@ -29,24 +29,50 @@
 public class DefaultApolloClientFactory implements ApolloClientFactory {
   @NonNull private final String streamRegistryUrl;
   @NonNull private final Consumer<ApolloClient.Builder> configurer;
+  private Credentials credentials;
 
   public DefaultApolloClientFactory(String streamRegistryUrl) {
     this(streamRegistryUrl, builder -> {});
   }
 
+  public DefaultApolloClientFactory(String streamRegistryUrl, Credentials credentials) {
+    this(streamRegistryUrl, builder -> {});
+    this.credentials = credentials;
+  }
+
+  public DefaultApolloClientFactory(String streamRegistryUrl, Consumer<ApolloClient.Builder> configurer, Credentials credentials) {
+    this(streamRegistryUrl, configurer);
+    this.credentials = credentials;
+  }
+
   @Override
   public ApolloClient create() {
-    var builder = builder()
-        .okHttpClient(new OkHttpClient.Builder().build());
+    OkHttpClient okHttpClient;
+
+    if (credentials != null) {
+      okHttpClient = new OkHttpClient.Builder()
+          .authenticator((route, response) -> response.request().newBuilder()
+              .header("Authorization", credentials.basic())
+              .build())
+          .addInterceptor(chain -> chain.proceed(chain.request().newBuilder()
+              .header("Authorization", credentials.basic())
+              .build()))
+          .build();
+    } else {
+      okHttpClient = new OkHttpClient.Builder().build();
+    }
+
+    var builder = builder().okHttpClient(okHttpClient);
     configurer.accept(builder);
```

Comment:
Because the `OkHttpClient` logic comes before the `configurer` it's possible the user could override it. Perhaps let's get rid of the configurer for now. We can deal with being able to configure other aspects of the `OkHttpClient` later if we need to.

Maybe something like:

```
    var okHttpClientBuilder = new OkHttpClient.Builder();
    if (credentials != null) {
      okHttpClientBuilder.addInterceptor(chain -> chain.proceed(chain.request().newBuilder()
          .header("Authorization", credentials.basic())
          .build()));
    }
    var okHttpClient = okHttpClientBuilder.build();

    return builder()
        .okHttpClient(okHttpClient)
        .serverUrl(streamRegistryUrl)
        .addCustomTypeAdapter(OBJECTNODE, new ObjectNodeTypeAdapter())
        .build();
```

Reply:
I would suggest something like the above too. There is no need for an 'else' branch in the construction this way. Either we inject the Interceptor or we don't.

Reply:
done